packages/noya-designsystem/src/components/InputField.tsx

```diff
@@ -64,35 +112,60 @@ function InputFieldLabel({ children = false }: InputFieldLabelProps) {
 const InputElement = styled(TextInput)<{
   labelPosition: LabelPosition;
   labelSize: number;
-}>(({ theme, labelPosition, labelSize }) => ({
-  ...theme.textStyles.small,
-  color: theme.colors.text,
-  width: '0px', // Reset intrinsic width
-  flex: '1 1 0px',
-  position: 'relative',
-  border: '0',
-  outline: 'none',
-  minWidth: '0',
-  textAlign: 'left',
-  alignSelf: 'stretch',
-  borderRadius: '4px',
-  paddingTop: '4px',
-  paddingBottom: '4px',
-  paddingLeft: labelPosition === 'start' ? `${6 + labelSize + 6}px` : '6px',
-  paddingRight: labelPosition === 'start' ? '6px' : `${6 + labelSize + 6}px`,
-  background: theme.colors.inputBackground,
-  '&:focus': {
-    boxShadow: `0 0 0 2px ${theme.colors.primary}`,
-  },
-}));
+  hasLabel: boolean;
+  hasDropdown: boolean;
+  textAlign?: Property.TextAlign;
+  disabled?: boolean;
+}>(
+  ({
+    theme,
+    labelPosition,
+    labelSize,
+    hasDropdown,
+    textAlign,
+    disabled,
+    hasLabel,
+  }) => ({
+    ...theme.textStyles.small,
+    color: disabled ? theme.colors.textDisabled : theme.colors.text,
+    width: '0px', // Reset intrinsic width
+    flex: '1 1 0px',
+    position: 'relative',
+    border: '0',
+    outline: 'none',
+    minWidth: '0',
+    textAlign: textAlign ?? 'left',
+    alignSelf: 'stretch',
+    borderRadius: '4px',
+    paddingTop: '4px',
+    paddingBottom: '4px',
+    paddingLeft:
+      hasLabel && labelPosition === 'start' ? `${6 + labelSize + 6}px` : '6px',
+    paddingRight:
+      labelPosition === 'start'
```

Comment:
We probably want to check `hasLabel` here first to get the correct behavior in the case of a dropdown with a `labelPosition` of `start`. Currently this would short-circuit and do `6px` even if there were a dropdown

Reply:
```css
paddingRight:
      (!hasLabel || !hasDropdown) && labelPosition === 'end'
        ? '6px'
        : `${6 + (hasLabel ? labelSize + 6 : 0) + (hasDropdown ? 11 : 0)}px`,
``` 

Like this ? 🤔

Reply:
I'm not sure 😂. I might try to simplify it after merging

Reply:
I'm going to update it to this

![Screen Shot 2021-06-02 at 1 24 40 PM](https://user-images.githubusercontent.com/1198882/120547510-12a93a80-c3a6-11eb-875b-59c30e36096f.png)